independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/core/Serialisers.java

```diff
@@ -91,11 +91,12 @@ private void readerLookup(MediaType mediaType, RuntimeType runtimeType, List<Med
             List<ResourceReader> goodTypeReaders) {
         if (goodTypeReaders != null && !goodTypeReaders.isEmpty()) {
             List<ResourceReader> mediaTypeMatchingReaders = new ArrayList<>(goodTypeReaders.size());
-            for (ResourceReader goodTypeReader : goodTypeReaders) {
+            for (int i = 0; i < goodTypeReaders.size(); i++) {
+                ResourceReader goodTypeReader = goodTypeReaders.get(i);
                 if (!goodTypeReader.matchesRuntimeType(runtimeType)) {
                     continue;
                 }
-                MediaType match = MediaTypeHelper.getBestMatch(mt, goodTypeReader.mediaTypes());
+                MediaType match = MediaTypeHelper.getBestMatch(mt, goodTypeReader.modifiableMediaTypes());
```

Comment:
So `MediaTypeHelper.getBestMatch` modifies its parameter? And `modifiableMediaTypes` provides a back-door to its immutable view? I can't recall the lifecycle of `ResourceReader/Writer`, but if it's application-long, it doesn't seem like a good idea to modify it while looking up reader/writers, no?

Probably I didn't understand something :)

Reply:
Yes, `bestMatch` does modify the parameters. But `modifiableMediaTypes` doesn't allow you to change the original order in any way